pkg/cmd/server/api/validation/node.go

```diff
@@ -140,3 +141,24 @@ func ValidateVolumeConfig(config api.NodeVolumeConfig, fldPath *field.Path) fiel
 	}
 	return allErrs
 }
+
+func ValidateCloudProviderNodeName(cloud cloudprovider.Interface, config api.NodeConfig) error {
```

Comment:
this doesn't belong in this package... move to an EnsureCloudProviderName() method on NodeConfig, like EnsureKubeletAccess